### PR TITLE
feat: Dynamic Dimension using ui-core components (DHIS2-8831)

### PR DESCRIPTION
**Pull request by martinkrulltott**
**TL;DR;**

1. Material UI replaced by `ui-core`
2. BREAKING CHANGE and known issues / missing features

--------
**Description** 
Implements [DHIS2-8831](https://jira.dhis2.org/browse/DHIS2-8831). The `DynamicDimension` used to be built on the `ItemSelector`. This has now been rebuilt to use [@dhis2/ui-core Transfer component](https://ui-core.dhis2.nu/#/api?id=transferproptypes-object) instead, to no longer depend on Material UI.

More details and full spec in the JIRA ticket.

Very similar to the changes made in https://github.com/dhis2/analytics/pull/452/

As `Transfer` takes care of all the item handling (select, deselect, reorder) internally, the `onDeselect` and `onReorder` props are no longer in use, as this functionality is now bundled within the `onSelect` prop, hence the breaking change.

Since this will be a breaking change anyway, I took the liberty to rename the `d2` and `dialogId` to something more semantic - `context` and `dimensionId` respectively. Feedback welcome!

--------
**BREAKING CHANGE** 
`DynamicDimension` props `onDeselect` and `onReorder` deprecated
`DynamicDimension` prop `d2` renamed to `context`
`DynamicDimension` prop `dialogId` renamed to `dimensionId`

--------
**Known issues**
- Loading spinner missing while fetching the items, causing an unwanted flash of empty state and "No items available" while loading. https://jira.dhis2.org/browse/TECH-379

--------
**Missing features**

- Right side title (_Selected Data_ in spec) missing. https://github.com/dhis2/ui/issues/90
- Right side styles (right side items are green in the spec) missing. https://github.com/dhis2/ui/issues/105
- Filter field placeholder (_Search_ in spec) missing. https://github.com/dhis2/ui/issues/131
- Filter field _dense_ prop missing. _Solution pending discussion_ (https://dhis2.slack.com/archives/CBM8LNEQM/p1591176539073400)

--------
**Screenshots**

_Default state, the dynamic dimension selector is now based on ui-core_ `Transfer` 
image

_The filter field can be used for searching_
image

### CODE REVIEW

File: src/components/DynamicDimension/ItemSelector.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+import styles from '../styles/DimensionSelector.style'
+import { TransferOption } from '../TransferOption'
+import GenericIcon from '../../assets/DimensionItemIcons/GenericIcon'
+import {
+    TRANSFER_HEIGHT,
+    TRANSFER_OPTIONS_WIDTH,
+    TRANSFER_SELECTED_WIDTH,
+} from '../../modules/dimensionSelectorHelper'
+
+class ItemSelector extends Component {
```
Comment (edoardo) [inline src/components/DynamicDimension/ItemSelector.js]: Why not making also this a functional component?

**Review by edoardo — APPROVED** [review]
LGTM

File: src/components/DynamicDimension/DynamicDimension.js [hunk 2]
```diff
@@ -1,139 +1,61 @@
...
+
+    useEffect(() => {
+        if (!items || !items.length) {
+            getItems()
+        }
+    }, [])
+
+    const getItems = async () =>
+        setItems(await apiFetchItemsByDimension(context, dimensionId))
+    // TODO: This needs to be refactored to use a loading spinner once Transfer supports it: https://jira.dhis2.org/browse/TECH-379
```
Comment (edoardo) [inline src/components/DynamicDimension/DynamicDimension.js:24]: This can be easily achieved with the data engine.
When looking into adding the spinner I think we should also replace d2 with data engine.

File: src/components/DynamicDimension/ItemSelector.js (hunk 1 above)
Comment (martinkrulltott) [reply src/components/DynamicDimension/ItemSelector.js]: fair point, fixed!
